service/resource/deployment/resource.go

```diff
@@ -67,23 +67,26 @@ type Resource struct {
 
 	// Settings.
 
-	uriVersion string
+	templateVersion string
 }
 
 // New creates a new configured deploy resource.
 func New(config Config) (*Resource, error) {
 	// Dependencies.
 	if config.AzureConfig == nil {
-		return nil, microerror.Maskf(invalidConfigError, "config.AzureConfig must not be empty.")
+		return nil, microerror.Maskf(invalidConfigError, "config.AzureConfig must not be empty")
 	}
 	if config.CertWatcher == nil {
 		return nil, microerror.Maskf(invalidConfigError, "config.CertWatcher must not be empty")
 	}
 	if config.CloudConfig == nil {
-		return nil, microerror.Maskf(invalidConfigError, "config.CloudConfig must not be empty.")
+		return nil, microerror.Maskf(invalidConfigError, "config.CloudConfig must not be empty")
 	}
 	if config.Logger == nil {
-		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty.")
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+	if config.TemplateVersion == "" {
```

Comment:
We usually differentiate between dependencies and settings here. Now the template version is validated with the dependencies but the template version is a setting.

Reply:
Done.